src/server.rs

```diff
@@ -241,13 +244,41 @@ struct Cli {
 }
 
 fn main() -> Result<(), Box<dyn std::error::Error>> {
-    let contents = fs::read_to_string("./config.toml")
-        .expect("Something went wrong reading the file");
-
-    let config: Config = toml::from_str(&contents)
-        .expect("Could not deserialize the config file");
-
-    println!("{:#?}", config);
+    // Attempt to read the contents of the config file
+    let contents = fs::read_to_string("../../config.toml");
+    match contents {
+        Ok(contents) => {
+            // Open a log file for writing
+            let mut log_file = File::create("../../config_parsing.log")?;
+
+            // Attempt to deserialize the config file
+            let config: Result<Config, _> = toml::from_str(&contents);
+            match config {
+                Ok(config) => {
+                    println!("{:#?}", config);
+                    // Write the parsed configuration to the log file
+                    writeln!(log_file, "{:#?}", config)?;
+                },
+                Err(err) => {
+                    // Log the error to the file
+                    writeln!(log_file, "Could not deserialize the config file: {:?}", err)?;
+                    // Print the error to stderr
+                    eprintln!("Could not deserialize the config file: {:?}", err);
```

Comment:
I think we can just have the error write down to the log file, as soon as config / logging works.

Reply:
removed stdout